src/oidcendpoint/session/manager.py

```diff
@@ -0,0 +1,290 @@
+        """
+        if user_id:
+            _user_id = user_id
+        elif session_id:
+            _user_id = unpack_session_key(session_id)[0]
+        else:
+            raise AttributeError("Must have one of user_id or session_id")
+
+        user_info = self.get([_user_id])
+
+        return user_info["authentication_event"]
+
+    def revoke_client_session(self, session_id: str):
+        """
+        Revokes a client session
+
+        :param session_id: Session identifier
+        """
+        parts = unpack_session_key(session_id)
+        if len(parts) == 2:
+            _user_id, _client_id = parts
+        elif len(parts) == 3:
+            _user_id, _client_id, _ = parts
+        else:
+            raise ValueError("Invalid session ID")
+
+        _info = self.get([_user_id, _client_id])
+        self.set([_user_id, _client_id], _info.revoke())
+
+    def revoke_grant(self, session_id: str):
+        """
+        Revokes the grant pointed to by a session identifier.
+
+        :param session_id: A session identifier
+        """
+        _path = unpack_session_key(session_id)
+        _info = self.get(_path)
+        _info.revoke()
+        self.set(_path, _info)
+
+    def grants(self, session_id: str) -> List[Grant]:
+        """
+        Find all grant connected to a user session
+
+        :param session_id: A session identifier
+        :return: A list of grants
+        """
+        uid, cid, _gid = unpack_session_key(session_id)
+        _csi = self.get([uid, cid])
+        return [self.get([uid, cid, gid]) for gid in _csi['subordinate']]
+
+    def get_session_info(self, session_id: str) -> dict:
+        _user_id, _client_id, _grant_id = unpack_session_key(session_id)
+        return {
+            "session_id": session_id,
+            "user_id": _user_id,
+            "client_id": _client_id,
+            "user_session_info": self.get([_user_id]),
+            "client_session_info": self.get([_user_id, _client_id]),
+            "grant": self.get([_user_id, _client_id, _grant_id])
+        }
+
+    def get_session_info_by_token(self, token_value: str) -> dict:
+        _token_info = self.token_handler.info(token_value)
+        return self.get_session_info(_token_info["sid"])
+
+    def add_grant(self, user_id: str, client_id: str, **kwargs) -> Grant:
+        """
+        Creates and adds a grant to a user session.
+
+        :param user_id: User identifier
+        :param client_id: Client identifier
+        :param kwargs: Keyword arguments to the Grant class initialization
+        :return: A Grant instance
+        """
+        args = {k: v for k, v in kwargs.items() if k in Grant.attributes}
+        _grant = Grant(**args)
+        self.set([user_id, client_id, _grant.id], _grant)
+        _client_session_info = self.get([user_id, client_id])
+        _client_session_info["subordinate"].append(_grant.id)
+        self.set([user_id, client_id], _client_session_info)
```

Comment:
This is done by `Database.set` (https://github.com/IdentityPython/oidcendpoint/pull/104/files#diff-93bc9f688c9bce53870f2c48de99c5e6b094668e243a710ce6d72abad9708a79R60)

Reply:
So your proposal is to remove lines 256-257 ?

Reply:
257-258